connection.go

```diff
@@ -280,25 +281,32 @@ func (conn *Connection) Handle() interface{} {
 }
 
 func (conn *Connection) dial() (err error) {
+	const unixSocketKey = `unix/:`
+	var connection net.Conn
+	network := `tcp`
+	address := conn.addr
 	timeout := conn.opts.Reconnect / 2
 	if timeout == 0 {
 		timeout = 500 * time.Millisecond
 	} else if timeout > 5*time.Second {
 		timeout = 5 * time.Second
 	}
-	connection, err := net.DialTimeout("tcp", conn.addr, timeout)
+	// Select tcp/ip or unix socket
+	if strings.Index(strings.ToLower(address), unixSocketKey) == 0 {
```

Comment:
If you say replace it with HasPreffix I will easily replace with HasPreffix
Reject this pull request